actor/context.go

```diff
@@ -38,24 +46,39 @@ type Context interface {
 	// the NotInfluenceReceiveTimeout interface, the timer will not be reset
 	SetReceiveTimeout(d time.Duration)
 
-	// ReceiveTimeout returns the current timeout
-	ReceiveTimeout() time.Duration
+	CancelReceiveTimeout()
 
-	// SetBehavior replaces the actors current behavior stack with the new behavior
-	SetBehavior(behavior ActorFunc)
+	// Forward forwards current message to the given PID
+	Forward(pid *PID)
 
-	// PushBehavior pushes the current behavior on the stack and sets the current Receive handler to the new behavior
-	PushBehavior(behavior ActorFunc)
+	AwaitFuture(f *Future, continuation func(res interface{}, err error))
+}
 
-	// PopBehavior reverts to the previous Receive handler
-	PopBehavior()
+type SenderContext interface {
+	// Message returns the current message to be processed
+	Message() interface{}
```

Comment:
Shouldn't the `Message()` be a part of the `ReceiverContext`? It seems strange to have a `Message()`on  RootContext. The same goes for the `MessageHeader()`

Reply:
I would opt for a separated message interface if possible. As we have sender and receiver that need to access messages we need that in a different place.